### PR TITLE
feat: make a support of Qt app to show Platform transfer Tx

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Transfers from platform have incorrectly shown amount in Dash Core wallet app.
They also shown in Qt app as self-send that is not completely true.

## What was done?
Added new type of transaction to Qt App, added a filter for its type, fixed calculation of output for tx records.
As well added a new type of transaction `platform-transfer` in rpc output of `gettransaction` RPC

## How Has This Been Tested?
Make a Platform Transfer transaction on RegTest and check it in Dash Core

image

Helper to see it: export dpath=/tmp/dash_func_test_PATHPATH/ ; src/qt/dash-qt -regtest -conf=$dpath/node0/dash.conf -datadir=$dpath/node0/ -debug=0 -debuglogfile=/dev/stdout

## Breaking Changes
There's new type of transaction "platform-transfer" in rpc output of `gettransaction`.

**This PR DOES NOT change any consensus rules.**
Breaking changes that makes withdrawal transaction immature is moved to https://github.com/dashpay/dash/pull/6128

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
b9118f5671d2bc9947f02dbeb0c6d3331be08e4f looks incomplete - you just import a key but you don't check any tx/rpc

**Comment by kxcd** [comment]
The term `withdrawal` will be confusing.  Suggest different name eg  `Asset pool` to denote it comes from an asset pool, or more specifically `Platform` to indicate this is coming from Platform.  It will be clear this a credit to the wallet, so no need to muddy the waters with terms like `withdrawal`.

**Comment by knst** [comment]
@thephez 
> The term withdrawal will be confusing.

Could you suggest? just "Asset unlock" is not good too.
 Internally we already call it asset unlock or withdrawal.

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
General NACK for v21.0; we are in feature freeze and releasing in 9 days. We need to avoid new features at all costs. We can include this in a fast following 21.1

**Comment by thephez** [comment]
> @thephez
> 
> > The term withdrawal will be confusing.
> 
> Could you suggest? just "Asset unlock" is not good too. Internally we already call it asset unlock or withdrawal.

Yes, we should think about this. I agree with @kxcd. It will be confusing because it is _not_ a withdrawal from the wallet's perspective - it's an incoming tx just like a masternode reward, incoming payment, etc. Maybe something like `platform transfer` or `asset/credit pool distribution`. If we used `* transfer`, it could also be used to show asset lock txs as well as unlocks. What do you guys think?

**Comment by kxcd** [comment]
Thanks Phez, I like `Platform Transfer` best.  Though I also like `asset/credit pool` and then let the wallet show you the direction since we can both send into Platform and receive from Platform, ideally, the name is the same for both transactions, just the direction will be different. The good thing about `Platform Transfer` is it can be in or out, it's directionless.

**Review by UdjinM6 — APPROVED** [review]
utACK 70cbef978ca808c74b00d378ca120317c8da7bb5 (mac-build CI failure is unrelated)

**Comment by PastaPastaPasta** [comment]
I want @thephez's approval for terminology / UX. I think I generally agree with him and @kxcd that "withdrawal" is confusing

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
general NACK on "withdrawal" terminology. Want utACK from @thephez / suggestions

**Comment by knst** [comment]
> general NACK on "withdrawal" terminology. Want utACK from @thephez / suggestions

current version has mentioning of `withdrawal` only in functional tests, Qt & rpc are updated already

**Comment by PastaPastaPasta** [comment]
> > general NACK on "withdrawal" terminology. Want utACK from @thephez / suggestions
> 
> current version has mentioning of `withdrawal` only in functional tests, Qt & rpc are updated already

Maybe add information about what the terms are then in description?

File: src/interfaces/wallet.h [hunk 1]
```diff
@@ -408,6 +408,7 @@ struct WalletTx
     int64_t time;
     std::map<std::string, std::string> value_map;
     bool is_coinbase;
+    bool is_platform_transfer;
```
Comment (PastaPastaPasta) [inline src/interfaces/wallet.h:411]: I know none of the others do it, but we should ensure these are always initialized
```suggestion
    bool is_platform_transfer{false};
```

**Comment by knst** [comment]
> @[knst](https://github.com/knst) knst [force-pushed](https://github.com/dashpay/dash/compare/70cbef978ca808c74b00d378ca120317c8da7bb5..3cd2115b17e8750a1519a64a158e19d5331aaba8) the feat-withdrawals-qt branch from[ 70cbef9 ](https://github.com/dashpay/dash/commit/70cbef978ca808c74b00d378ca120317c8da7bb5)to 3cd2115

there's extra fixes and improvements. Also PR description is updated and re-made a screenshot with "platform transfer" instead "withdrawal"

> @[knst](https://github.com/knst) knst [force-pushed](https://github.com/dashpay/dash/compare/3cd2115b17e8750a1519a64a158e19d5331aaba8..9748a048b41a83be9703bf76d69ca46f73330d31) the feat-withdrawals-qt branch from[ 3cd2115 ](https://github.com/dashpay/dash/commit/3cd2115b17e8750a1519a64a158e19d5331aaba8)to 9748a04

rebased on the top of current develop

**Review by thephez — COMMENTED** [review]
The "Platform transfer" terminology works for me :+1:

File: src/interfaces/wallet.h (hunk 1 above)
Comment (PastaPastaPasta) [reply src/interfaces/wallet.h]: please squash 9748a048b41a83be9703bf76d69ca46f73330d31 into a9a7f0766844543c90e249896f5ca10243c56c1d

Comment (PastaPastaPasta) [reply src/interfaces/wallet.h]: done in 9748a048b41...21f174aff10

**Review by PastaPastaPasta — APPROVED** [review]
utACK [21f174a](https://github.com/dashpay/dash/pull/6131/commits/21f174aff107f9aec9b758a00bde67b96ad3cd55)

**Pull request merged**